### PR TITLE
avoid invoking status when updating indicator for current repository

### PR DESCRIPTION
**Pull request by shiftkey**
## Overview

Related to #7464 

## Description

Whenever `AppStore._refreshRepository` runs (updating the current repository), it also runs `AppStore.refreshIndicatorsForRepositories` at the end.

Internally this does two expensive things:

 - run `AppStore._loadStatus()` to see whether there are working directory changes
 - run `AppStore.fetch()` to background fetch and detect the ahead/behind of the current branch

This feels rather wasteful to do for the active repository, and we already have a `AppStore._loadStatus()` result from earlier in `AppStore._refreshRepository` that has sufficient information for this UI (are there working changes?).

Given this function opts out of fetching while refreshing, and we have a relatively recent `getStatus` result, this PR introduces a more explicit `AppStore.updateSidebarIndicator` that is now used in `refreshRepository`, and leaves the existing behaviour in `AppStore.refreshAllSidebarIndicators` to be run when the background refresh happens.

This PR eliminates three Git operations that run on every refresh (a `git status` and two `git diff`).

On a `desktop/desktop` repository with a clean working directory:

**Before**

**After**

On a `nodejs/node` repository with 33k files removed in the working directory:

**Before**

**After**

## Release notes

Notes: no-notes

### CODE REVIEW

**Review by iAmWillShepherd — DISMISSED** [review]
My comment doesn't need to hold up this PR since what's here works just fine.

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]: i'm 👍 with this change, but we should also update this logic to do avoid implicit object boolean-ness: https://github.com/desktop/desktop/blob/e63ab5b33d6012d51a3e98b63563acdbcc59e3cb/app/src/lib/stores/app-store.ts#L2459

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]: can we make this an explicit boolean comparison too?

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]: i think we can call this right after `_loadStatus` with its result regardless of what it returns and then we can remove the `this.updateSidebarIndicator(repository, status)` at the end of this function entirely.

since `updateSidebarIndicator` is async and fairly straightforward i don't think there's risk there.

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [reply app/src/lib/stores/app-store.ts]: that _might_ be risky if there are a large number of repositories (since we'd flood the process with promises), so i think i have a slight preference for the original implementation.

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]: i like that this allows us to remove that extra boolean 💯

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]: i wish there was a clean way to use `updateSidebarIndicator` here without doing extra work since the logic is so similar but i can't quite figure out a way to do that right now

**Review by outofambit — CHANGES_REQUESTED** [review]
a couple tweaks, please, before we merge this. overall this is a nice refactor!

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [reply app/src/lib/stores/app-store.ts]: We talked about this in #5312 as there were some performance issue when we initially shipped the repository indicators - we'd create a number promises in the background to do work, and for users with dozens of repositories it would significantly affect things with no indicator of why.

A `for-loop` is fine here.

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [reply app/src/lib/stores/app-store.ts]: Yeah, I stared at this a bunch while I was in here and couldn't really tease out something I liked either. 

Gonna keep thinking about it, but I won't hold up this PR and the benefits it already gives.

Comment (outofambit) [reply app/src/lib/stores/app-store.ts]: > Gonna keep thinking about it, but I won't hold up this PR and the benefits it already gives.

yep, totally agree

**Review by outofambit — APPROVED** [review]

**Pull request merged**
